File: 0452-minimum-number-of-arrows-to-burst-balloons/0452-minimum-number-of-arrows-to-burst-balloons.cpp
class Solution {
public:
    int findMinArrowShots(vector<vector<int>>& p) {
        
        sort(p.begin(),p.end());
        
        int s = p[0][0],e = p[0][1];
        int count = 1;
        for(int i = 1; i<p.size(); i++){
            
            if(p[i][0] <= e && p[i][1] >= s){
                
                s = min(p[i][0],s);
                e = min(p[i][1],e);
            }else{
                count++;
                s = p[i][0];
                e = p[i][1];
            }
        }
        
        return count;
        
    }
};

File: 0452-minimum-number-of-arrows-to-burst-balloons/0452-minimum-number-of-arrows-to-burst-balloons.js
/**
 * @param {number[][]} points
 * @return {number}
 */
var findMinArrowShots = function(points) {
    
    points.sort((a,b) => a[1] - b[1]);
    
    console.log(points);
    let count = 1;
    // let s = points[0][0],e = points[0][1];
    let pos = points[0][1];
    for(let i = 1; i<points.length; i++){
        
        /*
        if(points[i][0] <= e && points[i][1] >= s){
            s = Math.min(s,points[i][0]);
            e = Math.min(e,points[i][1]);
        }else{
            count++;
            s = points[i][0];
            e = points[i][1];
            
        }
        */
        
        if(pos >= points[i][0])
            continue;
        count++;
        pos = points[i][1];
    }
    
    return count;
};